docs/automation/terraform/terraformGoogle.rst

```diff
@@ -0,0 +1,707 @@
+:lastproofread: 2024-03-25
+
+.. _terraformgoogle:
+
+Deploying VyOS in the google cloud
+==================================
+
+With the help of Terraform, you can quickly deploy VyOS-based infrastructure in the google cloud. If necessary, the infrastructure can be removed using terraform.
+Also we will make provisioning using Ansible.
+
+In this case, we'll create the necessary files for Terraform and Ansible next using Terraform we'll create a single instance on the google cloud and make provisioning using Ansible.
+
+Preparation steps for deploying VyOS on google 
+----------------------------------------------
+
+How to create a single instance and install your configuration using Terraform+Ansible+google 
+Step by step:
+
+google cloud
+
+
+  1 Create an account with google cloud and a new project
+
+.. image:: /_static/images/project.png
+   :width: 50%
+   :align: center
+   :alt: Network Topology Diagram
+
+  2 Create a service aacount and download your key (.JSON)
+
+.. image:: /_static/images/service.png
+   :width: 50%
+   :align: center
+   :alt: Network Topology Diagram
+
+.. image:: /_static/images/key.png
+   :width: 50%
+   :align: center
+   :alt: Network Topology Diagram
+
+The .JSON file download automaticly after creating and will look like:
+
+.. image:: /_static/images/json.png
+   :width: 50%
+   :align: center
+   :alt: Network Topology Diagram
+
+
+Terraform
+
+
+  1 Create an UNIX or Windows instance
+
+  2 Download and install Terraform
+
+  3 Create the folder for example /root/google
+
+.. code-block:: none
+
+   mkdir /root/google
+
+  4 Copy all files into your Terraform project "/root/google" (vyos.tf, var.tf, terraform.tfvars, .JSON), more detailed see `Structure of files Terrafom for google cloud`_
```

Comment:
please check the test build this will not render nice:
https://vyos--1345.org.readthedocs.build/en/1345/automation/terraform/terraformGoogle.html